docs/developer/dependency_resolution.md

```diff
@@ -0,0 +1,190 @@
+# Dependency resolution in the EDC
+
+The code base of the Eclipse Dataspace Connector is architected in away that allows for easily extending and swapping
+the core functionality using certain plug-points called _extensions_. One example would be to swap out an in-memory
+implementation for a datastore for one backed by an actual database. In order to achieve that there are several key
+components:
+
+- a service interface, typically located in an SPI module
+- a module providing the implementation, typically located in the `extensions` directory
+- the service registry, i.e. the `ServiceExtensionContext`. Since it is not quite an IoC container, we'll henceforth
+  refer to it as the "context".
+- a hook point into the loading sequence: an extension that that instantiates and registers the implementation class
+  with the context
+
+## Registering a service implementation
+
+As a general rule the module that provides the implementation also has to register it with
+the `ServiceExtensionContext` ("context" for short). This is done in an accompanying service extension. For example,
```

Comment:
nit: A bit redundant, you already mention few lines above _we'll henceforth
  refer to it as the "context"_